src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -6514,26 +6512,18 @@ TdsParserStateObject stateObj
                 stateObj._pendingData = true;
                 stateObj._messageStatus = 0;
 
-                SqlDataReader dtcReader = null;
                 stateObj.SniContext = SniContext.Snix_Read;
-                if (returnReader)
```

Comment:
Since returnReader is always false, else statement has been always executed, and if statement never runs. But this modification removed else statement instead of if statement, and it caused unexpected behavior...